pilot/pkg/leaderelection/leaderelection.go

```diff
@@ -136,3 +142,15 @@ func NewLeaderElection(namespace, name, electionID string, client kubernetes.Int
 		cycle: atomic.NewInt32(0),
 	}
 }
+
+// NewPerRevisionLeaderElection calls a leader election scoped to the replicas of a given revision.
+// For instance, in webhook patching, we want each revision to handle its own patching, but only a
+// single replica within that revision.
+func NewPerRevisionLeaderElection(namespace, name, revision, electionID string, client kubernetes.Interface) *LeaderElection {
```

Comment:
`NewLeaderElection` is already a general construction,  prefer reuse it instead of being aware of revision